### PR TITLE
Optimize type of params by Union Types

### PR DESCRIPTION
**Pull request by waitingsong**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by andy-ms** [comment]
We actually have a lint rule for this but linting isn't yet enabled for this package. You could do that by adding `types/ffi/tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

**Comment by waitingsong** [comment]
I have added that tslint.json and run lint, but got much check errors （with original code, not my change ）. Therefor, not commit it

**Comment by waitingsong** [comment]
I'll commit and push tslint.json file later.

File: types/ffi/tslint.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "extends": "dtslint/dt.json"
+}
```
Comment (andy-ms) [inline types/ffi/tslint.json]: You could add ignores for lint rules you don't want to handle right now:
```json
"rules": {
    "no-var-keyword": false,
    "no-duplicate-variable": false,
    ...
}
```

**Comment by RyanCavanaugh** [comment]
@waitingsong Please fix the failures indicated in the Travis CI log.

**Comment by waitingsong** [comment]
all fixed

**Comment by RyanCavanaugh** [comment]
@waitingsong - It appears Travis did not correctly run on this PR! /cc @RyanCavanaugh to investigate and advise.

**Comment by waitingsong** [comment]
@RyanCavanaugh  do you mean commit 821b8c8 ?

**Comment by waitingsong** [comment]
It works ok with commit 821b8c8 under my project. But i have not seen the format of function definition with colon such as
export const Library: {}   //

**Comment by waitingsong** [comment]
how's go on?

**Pull request merged**